src/main/java/bio/terra/cli/businessobject/resources/AiNotebook.java

```diff
@@ -118,15 +120,20 @@ public String resolve() {
   }
 
   /** Query the cloud for information about the notebook VM. */
-  public Instance getInstance() {
+  public Optional<Instance> getInstance() {
     InstanceName instanceName =
         InstanceName.builder()
             .projectId(projectId)
             .location(location)
             .instanceId(instanceId)
             .build();
     GoogleAiNotebooks notebooks = new GoogleAiNotebooks(Context.requireUser().getUserCredentials());
-    return notebooks.get(instanceName);
+    try {
+      return Optional.of(notebooks.get(instanceName));
+    } catch (SystemException ex) {
+      logger.error("Caught exception looking up notebook instance", ex);
+      return Optional.empty();
```

Comment:
`getInstance` should succeed regardless of whether `notebooks.get` does. Instead of changing `notebooks.get` to handle a specific exception, I'll change `getInstance` to handle all exceptions.